File: 2216-delete-the-middle-node-of-a-linked-list/2216-delete-the-middle-node-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {

    ListNode ptr=head;
    ListNode slow=head;
    ListNode fast=head;
    ListNode pre=null;
    if(head==null||head.next==null)
    return null;
    while(fast!=null&&fast.next!=null)
    {
        pre=slow;
    slow=slow.next;
    fast=fast.next.next;

    }
   /* if(slow.next!=null){
   slow.val=slow.next.val;
   slow.next=slow.next.next;
   }
   else
   {

    slow=null;
   }*/

  pre.next=slow.next;

   return head;
    }
}

File: 2216-delete-the-middle-node-of-a-linked-list/2216-delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {

    ListNode *ptr=head;
     ListNode *slow=head;
        ListNode *pre=head;
      ListNode *fast=head;
      if(head==NULL)
      return NULL;
       if(head->next==NULL)
      return 0;

    while(fast !=NULL && fast->next!=NULL )
{
    pre=slow;
 slow=slow->next;
 fast=fast->next->next;
 

}
pre->next=slow->next;
delete slow;
  return head;
    }
};